### DISCUSSION TITLE
switching track switched

### DISCUSSION THREAD

**Discussion by lukeashworth54** [community/community#56027] [Programming Help]
### Hi everyone,

hi all,
Luke here,

new to all of this programming,

I am trying to set up my Audino to use an automatic switch, a switch, on my layout.
i am using standard insulfrog switch.
I am using a mega 2560 with l298p motor shield.
also using 2 infra-red sensors with breadboard and a DR60030 servo.
I have tried to configure it with ai sketches and i till cannot get it to work.
is there someone able to help me with this project and the correct way of putting it together.
Please email me.
I have all pos and negatives in the correct places,
I have servo on digital pin 9
I have ir sensor on digital pins 3 and 4
and it still not working correctly can anyone help please.

and the code that i am using is below.

#include <Servo.h>

// Constants for infrared sensor pins
const int IR_SENSOR_1_PIN = 2;
const int IR_SENSOR_2_PIN = 3;

// Constants for servo control
const int SERVO_PIN = 4;
const int SERVO_ANGLE_CLOSED = 0;
const int SERVO_ANGLE_OPEN = 90;

// Variables to store infrared sensor states
int irSensor1State = 0;
int irSensor2State = 0;

// Initialize servo object
Servo trackSwitch;

void setup() {
// Initialize infrared sensor pins as inputs
pinMode(IR_SENSOR_1_PIN, INPUT);
pinMode(IR_SENSOR_2_PIN, INPUT);

// Initialize servo pin
trackSwitch.attach(SERVO_PIN);

// Set initial position of the track switch
trackSwitch.write(SERVO_ANGLE_CLOSED);
}

void loop() {
// Read the state of infrared sensors
irSensor1State = digitalRead(IR_SENSOR_1_PIN);
irSensor2State = digitalRead(IR_SENSOR_2_PIN);

if (irSensor1State == HIGH && irSensor2State == HIGH) {
// Both sensors detect a train, switch the track
trackSwitch.write(SERVO_ANGLE_OPEN);
delay(1000); // Adjust the delay time to ensure the servo has enough time to switch the track
} else {
// No train detected, close the track
trackSwitch.write(SERVO_ANGLE_CLOSED);
delay(1000); // Adjust the delay time to ensure the servo has enough time to switch the track
}
}

**Comment by samus-aran** [comment]
Welcome to the GitHub Community, @lukeashworth54 , we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!
